packages/rest/src/parser.ts

```diff
@@ -106,22 +103,23 @@ function getParamFromRequest(
     case 'header':
       // @jannyhou TBD: check edge cases
       return request.headers[spec.name.toLowerCase()];
-      break;
     // TODO(jannyhou) to support `cookie`,
     // see issue https://github.com/strongloop/loopback-next/issues/997
     default:
       throw RestHttpErrors.invalidParamLocation(spec.in);
   }
 }
 
+/**
+ * This method is mostly used for unit testing where Express is not set up
+ * to parse query string into `request.query` object
```

Comment:
IIRC, most of our tests are using Shot-based mock to build the request/response objects.

https://github.com/strongloop/loopback-next/blob/cb57a4211f098fce68cfa5e534bd2df3d84eb64f/packages/testlab/src/shot.ts#L108-L109

Can you please update the code building the mock to parse the query string the same way as Express (LoopBack4) does, and then remove `ensureRequestQueryWasParsed` completely?

Reply:
Fixed